File: 3784-minimum-deletion-cost-to-make-all-characters-equal/3784-minimum-deletion-cost-to-make-all-characters-equal.java
class Solution {
    public long minCost(String s, int[] cost) {
        long[] arr=new long[26];
        long sum=0,m=-1;
        for(int i=0;i<s.length();i++){
            arr[s.charAt(i)-97]+=cost[i];
            sum+=cost[i];
            m=Math.max(m,arr[s.charAt(i)-97]);
        }
        return sum-m;
    }
}

File: 3784-minimum-deletion-cost-to-make-all-characters-equal/3784-minimum-deletion-cost-to-make-all-characters-equal.cpp
class Solution {
public:
    long long minCost(string s, vector<int>& cost) {
        vector<long long> arr(26);
        long long sum=0,m=-1;
        for(int i=0;i<s.size();i++){
            arr[s[i]-97]+=cost[i];
            sum+=cost[i];
            m=max(m,arr[s[i]-97]);
        }
        return sum-m;
    }
};